hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaUtils.java

```diff
@@ -116,9 +116,26 @@ public static String getAvroRecordQualifiedName(String tableName) {
     return "hoodie." + sanitizedTableName + "." + sanitizedTableName + "_record";
   }
 
+
+  /**
+   * Validate whether the {@code targetSchema} is a valid evolution of {@code sourceSchema}.
+   * Basically {@link #isCompatibleProjectionOf(Schema, Schema)} but type promotion in the
+   * opposite direction
+   */
+  public static boolean isValidEvolutionOf(Schema sourceSchema, Schema targetSchema) {
```

Comment:
unit test for as many evolution scenarios as possible.

Reply:
@jonvex , can we do 2 kinds of evolution in a single batch eg., drop column and nested type evolution ? I think we need to test these scenarios so we can set the correct expectation in the product

Reply:
We have tests. We can do everything in the same batch